src/components/islands/recent-searches.jsx

```diff
@@ -1,32 +1,35 @@
-import { useEffect } from "react";
-import { useStore } from "@nanostores/react";
+import { useEffect, useState } from 'react';
+import { useStore } from '@nanostores/react';
 import { $recentSearches } from "../../lib/stores/search.js";
 
-/**
- * Recent Searches Component - An Island that displays a user's last 5 searches
- * 
- * @todo implement a default list instead of `null` when no `$recentSearch` is found 
- * @todo implement loading component to avoid flickering UI
- */
 export default function RecentSearches() {
   const recentSearches = useStore($recentSearches);
-
+  
   useEffect(() => {
-    $recentSearches.set({...JSON.parse(localStorage.getItem("jargons.dev:recent_searches"))})
+    const savedSearches = JSON.parse(localStorage.getItem("jargons.dev:recent_searches"));
+    if (savedSearches) {
+      $recentSearches.set(savedSearches);
+    }
+  
+    const loader = document.querySelector(".loading");
+    if (loader) loader.remove();
+
   }, []);
 
+
+
   return Object.values(recentSearches).length ? (
     <div className="space-y-3 ml-2 mt-4 md:mt-6">
-			<h2 className="text-2xl md:text-4xl font-black">Recent</h2>
-			<ol className="space-y-1.5 underline">
+      <h2 className="text-2xl md:text-4xl font-black">Recent</h2>
+      <ol className="space-y-1.5 underline">
         {Object.values(recentSearches).slice(0, 5).map((item, i) => (
           <li key={i}>
             <a href={item.url}>
-              { item.word }
+              {item.word}
             </a>
           </li>
         ))}
-			</ol>
-		</div>
+      </ol>
+    </div>
   ) : null;
```

Comment:
come into this `null` block. 😉

Reply:
Okay... But if the local storage is empty, the loading component will remain on the page

Reply:
Oh my! that is true 🤔 

So, we could use an Effect to understand when the page is loaded then kick the `LoadingComponent` or `RecentSearchComponent` or `null` on that, something like....

```js
const [ready, setReady] = useState(false);
useEffect(() => {
  setReady(true);
}, []);

if (!ready) {
  return <LoadingComponent />
}

return Object.values(recentSearches).length ? (
  <div className="space-y-3 ml-2 mt-4 md:mt-6">
    <h2 className="text-2xl md:text-4xl font-black">Recent</h2>
    <ol className="space-y-1.5 underline">
      {Object.values(recentSearches).slice(0, 5).map((item, i) => (
        <li key={i}>
          <a href={item.url}>
            {item.word}
          </a>
        </li>
      ))}
    </ol>
  </div>
) : null;
```

Thinking out loud here 😉